src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -288,15 +288,23 @@ private static function hasUserChanged(UserInterface $originalUser, TokenInterfa
         }
 
         if ($originalUser instanceof PasswordAuthenticatedUserInterface || $refreshedUser instanceof PasswordAuthenticatedUserInterface) {
-            if (!$originalUser instanceof PasswordAuthenticatedUserInterface || !$refreshedUser instanceof PasswordAuthenticatedUserInterface || $originalUser->getPassword() !== $refreshedUser->getPassword()) {
+            if (!$originalUser instanceof PasswordAuthenticatedUserInterface || !$refreshedUser instanceof PasswordAuthenticatedUserInterface) {
                 return true;
             }
 
+            if (null !== ($originalPassword = $originalUser->getPassword())
+                && ($refreshedPassword = $refreshedUser->getPassword()) !== $originalPassword
+                && (16 !== \strlen($originalPassword) || hash('xxh3', $refreshedPassword) !== $originalPassword)
```

Comment:
here is the meant of the PR: accepting null or hashed passwords in addition to the password itself

Reply:
You should check that `$refreshedPassword` is not `null` before calling `hash`.

Reply:
Updated by using $originalPassword as a fallback, which means invalidating the session (as is the case currently)
Not sure which scenario would allow this to happen but at least we have a defined behavior :)